src/czar/CzarConfig.h

```diff
@@ -44,15 +49,33 @@ namespace lsst::qserv::czar {
  */
 class CzarConfig {
 public:
-    CzarConfig(std::string configFileName) : CzarConfig(util::ConfigStore(configFileName)) {}
+    /**
+     * Create an instance of CzarConfig and load parameters from the specifid file.
+     * @note One has to call this method at least once before trying to obtain
+     *   a pointer of the instance by calling 'instance()'. The method 'create()'
+     *   can be called many times. A new instance would be created each time and
+     *   stored within the class.
+     * @param configFileName - path to worker INI configuration file
+     * @return the shared pointer to the configuration object
+     */
+    static std::shared_ptr<CzarConfig> create(std::string const& configFileName);
+
+    /**
+     * Get a pointer to an instance that was created by the last call to
+     * the method 'create'.
+     * @return the shared pointer to the configuration object
+     * @throws std::logic_error when attempting to call the bethod before creating an instance.
+     */
+    static std::shared_ptr<CzarConfig> instance();
```

Comment:
I think having `create()` get called more than once should probably be an error. Having 2 different instances of the configuration that do not match could be extremely confusing.

Reply:
Good point! I will fix that.